081-unregistration-of-KRaft-nodes.md

```diff
@@ -0,0 +1,117 @@
+For example, when you upgrade your cluster to a new Kafka minor version and try to also bump the metadata version, the metadata version update will be refused because the removed nodes that are still registered with the old Kafka version and do not support the new metadata version.
+
+Strimzi currently doesn't support the node unregistration.
+This has two main reasons:
+* Apache Kafka did not bother to properly document this process and we found out about it only by coincidence.
+* Apache Kafka has only limited support for unregistering nodes.
+  While the Kafka Admin API supports unregistration of the nodes, it does not let you list the registered nodes.
+  So you do not know what nodes should be unregistered.
+  This is causing a problems to stateless management tools such as Strimzi operators who might not reliably know what nodes existed before.
+  The proper solution for this is part of the [KIP-1073](https://cwiki.apache.org/confluence/display/KAFKA/KIP-1073%3A+Return+inactive+observer+nodes+in+DescribeQuorum+response) that is currently under discussion.
+
+## Motivation
+
+Despite the missing APIs being an important limitation of the KRaft implementation, Apache Kafka does not plan to address this in Apache Kafka 3.x releases.
+But this limitation raised a lot of discussions and concerns in the Strimzi community after we added it to our list of known KRaft limitations.
+So this proposal tries to address it in an alternative way that is not dependent on Apache Kafka and its Admin API.
+
+## Proposal
+
+This proposal explains the initial workaround that can be implemented purely on the Strimzi level to handle node unregistration.
+It also outlines the future / long-term solution that will be used once the implementation of this feature is completed in Apache Kafka.
+
+### Initial workaround implementation
+
+In order to be able to unregister the removed Kafka nodes, we need to be aware of the nodes that were removed.
+Strimzi Cluster Operator is aware of the scale-downs that it directly executed.
+These are the scale-downs resulting from changing the `.spec.replicas` field in the `KafkaNodePool` resources.
+But this information is not persisted anywhere and will not survive a crash or restart of the operator or an reconciliation failure that might happen between the nodes being scaled down and the node unregistration.
+Strimzi Cluster Operator is also not directly involved in some scale-down operations such as when a KafkaNodePool resource is deleted.
+In this case, the scale-down is done by Kubernetes garbage collection.
+
+To work around this limitation, the `Kafka` custom resource will be extended and will store a full list of the node IDs in its status section.
+During the reconciliation, this field will be used to get the information about the previously existing nodes.
+It will be compared with the nodes that are still in use the removed nodes will be unregistered.
```

Comment:
"It will be compared with the nodes that are still in use .... " is there something missing before the rest of the sentence? i.e " ... and the removed nodes will be unregistered"?